pkgs/tools/misc/jaeger/default.nix

```diff
@@ -0,0 +1,178 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, buildGoModule
+, git
+, gzip
+, nodejs-16_x
+, yarn
+}:
+
+let
+  version = "1.41.0";
+  src = fetchFromGitHub {
+    owner = "jaegertracing";
+    repo = "jaeger";
+    rev = "v${version}";
+    sha256 = "aa0RdEMRuKrGkXlxkMpouB/c924xRc4m6XTqISX2vkQ=";
+    fetchSubmodules = true;
+    leaveDotGit = true;
+  };
+
+  nodejs = nodejs-16_x;
+  yarn' = yarn.override { inherit nodejs; };
+
+  buildJaeger = { name, description, linkUI ? false }:
+    buildGoModule rec {
+      pname = "jaeger-${name}";
+      inherit version src;
+
+      vendorHash = "sha256-4vja9pI+t8mIF3+f8u5m+gTgC48/HdhEB7sJMTCYz0A=";
+      subPackages = [ "cmd/${name}" ];
+
+      ldflags = [
+        "-s"
+        "-w"
+        "-X github.com/jaegertracing/jaeger/pkg/version.latestVersion=v${version}"
+      ];
+
+      tags = lib.optional linkUI "ui";
+
+      nativeBuildInputs = [ git ];
+
+      preBuild = ''
+        GIT_SHA=$(git rev-parse HEAD)
+        DATE=$(TZ=UTC0 git show --quiet --date='format-local:%Y-%m-%dT%H:%M:%SZ' --format="%cd")
+        ldflags+=" -X github.com/jaegertracing/jaeger/pkg/version.commitSHA=$GIT_SHA"
+        ldflags+=" -X github.com/jaegertracing/jaeger/pkg/version.date=$DATE"
+      '' + lib.optionalString linkUI ''
+        cp --no-preserve=mode -r ${jaeger-ui}/* cmd/query/app/ui/actual/
+        find cmd/query/app/ui/actual -type f | grep -v .gitignore | xargs gzip --no-name
+      '';
+
+      postInstall = ''
+        mv $out/bin/${name} $out/bin/${pname}
+      '';
+
+      meta = with lib; {
+        inherit description;
+        homepage = "https://www.jaegertracing.io";
+        license = licenses.asl20;
+        maintainers = with maintainers; [ emilytrau ];
+      };
+    };
+
+  yarnCache = stdenv.mkDerivation {
```

Comment:
This is a little bit of an unusual pattern, so some comments explaining this derivation would be nice